subsys/bluetooth/mesh/dk_prov.c

```diff
@@ -160,6 +160,31 @@ static const struct bt_mesh_prov prov = {
 
 const struct bt_mesh_prov *bt_mesh_dk_prov_init(void)
 {
+	/* Generate an RFC-4122 version 4 compliant UUID.
+	 * Format:
+	 *
+	 * 0                   1                   2                   3
+	 * 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+	 * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	 * |                          time_low                             |
+	 * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	 * |       time_mid                |         time_hi_and_version   |
+	 * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	 * |clk_seq_hi_res |  clk_seq_low  |         node (0-1)            |
+	 * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	 * |                         node (2-5)                            |
+	 * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	 *
+	 * Where the 4 most significant bits of time_hi_and_version shall be
+	 * 0b0010 and the 2 most significant bits of clk_seq_hi_res shall be
+	 * 0b10. The remaining fields have no required values, and are fetched
+	 * from the HW info device ID. The fields are encoded in big endian
+	 * format.
+	 *
+	 * https://tools.ietf.org/html/rfc4122
+	 */
 	hwinfo_get_device_id(dev_uuid, sizeof(dev_uuid));
+	dev_uuid[6] = (dev_uuid[6] & BIT_MASK(4)) | BIT(5);
```

Comment:
Setting BIT(5) will create version 2 UUID, if version 4 is desired, BIT(6) should be set.

Reply:
Looks like you're right, although this is the same as is done in [the nRF5 SDK for mesh](https://github.com/NordicSemiconductor/nRF5-SDK-for-Mesh/blob/master/mesh/core/src/nrf_mesh_configure.c#L62)

Reply:
It is not fully equivalent. There, code results in : `x = (x & ~(0xF0)) | (0x04<<6) = (x & 0x0F) | (0x40)`, here it looks like it is `(x & 0x0F) | (0x20)`.

Reply:
You're right! Fooled by the macros, I mixed up the variant and version defines